src/expr/dtype.cpp

```diff
@@ -171,7 +171,7 @@ bool DType::resolve(const std::map<std::string, TypeNode>& resolutions,
 
   d_involvesExt = false;
   d_involvesUt = false;
-  for (const std::shared_ptr<DTypeConstructor> ctor : d_constructors)
+  for (const std::shared_ptr<DTypeConstructor>& ctor : d_constructors)
```

Comment:
Add the `const&` in the loop before as well.

Reply:
I don't think we can, since we are actually changing members of `ctor`, e.g. `ctor->d_constructor.setAttribute(DTypeIndexAttr(), index)`.

Reply:
The `const&` would only be about the `std::shared_ptr` itself, not what it points to. (It's not `std::shared_ptr<const DTypeConstructor>`...)